lithops/worker/handler.py

```diff
@@ -204,10 +205,32 @@ def run_task(task):
         jobrunner = JobRunner(task, jobrunner_conn, internal_storage)
         logger.debug('Starting JobRunner process')
         jrp = Process(target=jobrunner.run) if is_unix_system() else Thread(target=jobrunner.run)
+
+        # Start System monitoring if log level is set to DEBUG
+        if task.log_level == logging.DEBUG:
+            sys_monitor = SystemMonitor()
+            sys_monitor.start()
+
         jrp.start()
         jrp.join(task.execution_timeout)
+
+        if task.log_level == logging.DEBUG:
+            sys_monitor.stop()
         logger.debug('JobRunner process finished')
 
+        # Get and log System statistics if log level is DEBUG
+        if task.log_level == logging.DEBUG:
+            cpu_usage, cpu_system_time, cpu_user_time = sys_monitor.calculate_cpus_values()
+
+            call_status.add('worker_func_cpu_usage', cpu_usage)
+            call_status.add('worker_func_cpu_system_time', round(cpu_system_time, 8))
+            call_status.add('worker_func_cpu_user_time', round(cpu_user_time, 8))
+            call_status.add('worker_func_cpu_total_time', round(cpu_system_time + cpu_user_time, 8))
+
+            net_io = sys_monitor.get_network_io()
+            call_status.add('worker_func_sent_net_io', net_io[0])
+            call_status.add('worker_func_recv_net_io', net_io[1])
+
```

Comment:
Looks better now, thanks for the changes.

Since now it does not run every second, I was thinking about the possibility to remove `if task.log_level == logging.DEBUG:` and return this stats always, if the overhead of this code is negligible. So, can you check what is the overhead in seconds of this block of code? I think you only need to add a `logger.info("---to be removed---")` at line 233 and see the difference in time between it and the `logger.debug('JobRunner process finished')` in the functions logs.